cms/djangoapps/contentstore/views/preview.py

```diff
@@ -194,16 +194,15 @@ def _prepare_runtime_for_preview(request, block, field_data):
         # stick the license wrapper in front
         wrappers.insert(0, partial(wrap_with_license, mako_service=mako_service))
 
-    preview_anonymous_user_id = 'student'
+    anonymous_user_id = deprecated_anonymous_user_id = 'student'
     if individualize_anonymous_user_id(course_id):
```

Comment:
I guess dealing with that is not worth adding to this PR in terms of risk either way.

Reply:
It seems to have been added on purpose: https://github.com/openedx/edx-platform/pull/30248.

Reply:
Yeah, okay, good to not touch... 😛

Reply:
Misc. note, but now I think I better understand why the anonymous ID parsing in capa content that I've seen has something like this:

```python
import random
try:
	randomseed = int(anonymous_student_id, 16)
except:
	randomseed = 1
random.seed(randomseed)
```

... so it doesn't blow up the problem in preview where it can't parse because the strings are not hex strings. There's no need to change this behavior–I just put this note here because it was something that was puzzling me.

Reply:
FWIW, the motivation for that is that they want to have a handful of random variables that are generated the same way for a given student across multiple problems so that they can tie them together. Like if problem 1 asks you to calculate your monthly mortgage payments based on some variables, and then problem 2 asks you what those payments would be if you paid made a $100K payment after the first year, etc.

Reply:
I think it would actually be a very useful feature if we could assign variables to some namespace that lives across problems and let XBlocks access that–people have basically hacked that together in a few different ways over the years. Maybe as an XBlock service some day, though we'd have to think through the implications pretty critically.

Reply:
> Unless I'm missing something, the hash function works the same way as int(hex_string, 16) for hex values. It works for non-hex strings, too.

Python's builtin `hash` function is only guaranteed to return consistently within a given run of a Python process, so it would be a bad fit for deriving anything stable.

```
edx-platform on  ormsbee-anonymous-id-xb-service via 🐍 v3.8.12 (tutor_nightly) is 📦 v0.1.0 took 12s
❯ python
Python 3.8.12 (default, Dec 10 2021, 16:41:45)
[Clang 13.0.0 (clang-1300.0.29.3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> hash("Hello World!")
6517261053207716808
>>> hash("Hello World!")
6517261053207716808
>>>

edx-platform on  ormsbee-anonymous-id-xb-service via 🐍 v3.8.12 (tutor_nightly) is 📦 v0.1.0 took 2m18s
❯ python
Python 3.8.12 (default, Dec 10 2021, 16:41:45)
[Clang 13.0.0 (clang-1300.0.29.3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> hash("Hello World!")
740855705488943589
>>> hash("Hello World!")
740855705488943589
>>>
```